src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -302,6 +302,31 @@ private void evaluateSystemIndicesAccess(
                 }
         }
 
+        if (user.isPluginUser()) {
+            Set<String> matchingSystemIndices = SystemIndexRegistry.matchesPluginSystemIndexPattern(
+                user.getName().replace("plugin:", ""),
+                requestedResolved.getAllIndices()
+            );
+            if (requestedResolved.getAllIndices().equals(matchingSystemIndices)) {
```

Comment:
`requestedResolved.getAllIndices()` must be guarded by a `isLocalAll()`, otherwise you will get wrong results for `_all`  requests.

Reply:
See response below. Can the whole top level condition be `if (user.isPluginUser() && !requestedResolved.isLocalAll()) { ... }`?